### PR TITLE
MM-20625 Unit Tests

### PR DESCRIPTION
**Pull request by imisshtml**
#### Summary
Unit tests for Image loader

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-20625

#### Checklist
- [x] Added or updated unit tests (required for all new features)

#### Device Information
This PR was tested on: iPhone X Simulator 12.2

### CODE REVIEW

File: app/components/progressive_image/progressive_image.test.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import {shallow} from 'enzyme';
+import {shallowWithIntl} from 'test/intl-test-helper';
```
Comment (enahum) [inline app/components/progressive_image/progressive_image.test.js]: why this change? as far as I can tell the ProgressiveImage component does not use intl

File: app/components/progressive_image/progressive_image.test.js [hunk 2]
```diff
@@ -22,22 +22,34 @@ describe('ProgressiveImage', () => {
...
-        const props = {
-            ...baseProps,
-            thumbnailUri: null,
-        };
+    test('should load image', () => {
+        const wrapper = shallowWithIntl(<ProgressiveImage {...baseProps}/>);
+        const instance = wrapper.instance();
+        const load = jest.spyOn(instance, 'load');
+        load();
+        expect(load).toHaveBeenCalled();
```
Comment (enahum) [inline app/components/progressive_image/progressive_image.test.js]: you are calling load explicitly in the above line, so it seems obvious is going to get called. @saturninoabril can you provide assistance in how he can test that `load` is being called in `componentDidMount`?

File: app/components/progressive_image/progressive_image.test.js [hunk 3]
```diff
@@ -22,22 +22,34 @@ describe('ProgressiveImage', () => {
...
-        const wrapper = shallow(
-            <ProgressiveImage {...props}/>
-        );
-        expect(wrapper.getElement()).toMatchSnapshot();
+    test('should set image state', () => {
+        const wrapper = shallowWithIntl(<ProgressiveImage {...baseProps}/>);
+        const instance = wrapper.instance();
+        const setImage = jest.spyOn(instance, 'setImage');
+        setImage('test.png');
+        expect(setImage).toHaveBeenCalled();
```
Comment (enahum) [inline app/components/progressive_image/progressive_image.test.js]: same as above, but in this case I don't think you need to check that it was called.

In this test you should check that `setImage` is called after ImageCacheManager.cache executes the callback.

Comment (saturninoabril) [reply app/components/progressive_image/progressive_image.test.js]: Can be,
```javascript
jest.useFakeTimers();

describe('ProgressiveImage', () => {
...
    test('should set image when imageUri is set but not the thumbnailUri', () => {
        const wrapper = shallow(
            <ProgressiveImage
                {...baseProps}
                thumbnailUri={null}
            />
        );
        const instance = wrapper.instance();
        jest.spyOn(instance, 'setImage');

        instance.componentDidMount();
        jest.runAllTimers();
        expect(instance.setImage).toHaveBeenCalledTimes(1);
    });
```

File: app/components/progressive_image/progressive_image.test.js [hunk 4]
```diff
@@ -22,22 +22,34 @@ describe('ProgressiveImage', () => {
...
+        expect(wrapper.state().uri).toMatch('test.png');
+    });
+
+    test('should set thumbnail state', () => {
+        const wrapper = shallowWithIntl(<ProgressiveImage {...baseProps}/>);
+        const instance = wrapper.instance();
+        const setThumbnail = jest.spyOn(instance, 'setThumbnail');
+        setThumbnail('test.png');
+        expect(setThumbnail).toHaveBeenCalled();
+        expect(wrapper.state().thumb).toMatch('test.png');
```
Comment (saturninoabril) [inline app/components/progressive_image/progressive_image.test.js]: Can be,
```javascript
jest.useFakeTimers();

describe('ProgressiveImage', () => {
...
    test('should set thumbnail when thumbnailUri is set', () => {
        const wrapper = shallow(
            <ProgressiveImage
                {...baseProps}
                imageUri={null}
            />
        );
        const instance = wrapper.instance();
        jest.spyOn(instance, 'setThumbnail');

        jest.runAllTimers();
        expect(instance.setThumbnail).toHaveBeenCalledTimes(1);
    });
```

**Comment by imisshtml** [comment]
Updated and tested based on comments, thanks for the help everyone :)

**Pull request merged**